programs/vote/src/vote_state/mod.rs

```diff
@@ -530,13 +530,21 @@ impl VoteState {
                             // Then filter it out
                             vote_state_update_indexes_to_filter.push(vote_state_update_index);
                         }
-                        vote_state_update_index += 1;
+                        if check_root.is_some() {
+                            check_root = None;
```

Comment:
This implies the root is smaller than the smallest SlotAncestor, which can't happen due to the check here right: https://github.com/solana-labs/solana/blob/c463ee29f0284aebbf05b0bcbf04e8f3b067b762/programs/vote/src/vote_state/mod.rs#L475. We can assert this case is not true.

Reply:
It can happen if the vote state's current root is too old to be in slot hashes history itself.

Reply:
e.g `test_check_update_vote_state_older_than_history_root`

Reply:
ah I see, thanks. can we add an assert here then that `self.root_slot < earliest_slot_hash_in_history` and add a comment describing why this is true?